Sources/Quick/Example.swift

```diff
@@ -90,12 +90,23 @@ final public class Example: _ExampleBase {
             #else
             let file = callsite.file
             #endif
+            #if swift(>=5.3) && !SWIFT_PACKAGE
+            let location = XCTSourceCodeLocation(filePath: file, lineNumber: Int(callsite.line))
+            let sourceCodeContext = XCTSourceCodeContext(location: location)
+            let issue = XCTIssue(
+                type: .thrownError,
+                compactDescription: description,
+                sourceCodeContext: sourceCodeContext
+            )
+            QuickSpec.current.record(issue)
```

Comment:
That's because on Linux's XCTest (swift-corelibs-xctest) `record(_:)` method is not implemented yet.

https://github.com/apple/swift-corelibs-xctest

Reply:
That makes sense. But how does the `SWIFT_PACKAGE` macro relate to being on Linux or not?

I would have expected `!os(Linux)`

Reply:
`os(Linux)` is not preferable as there are other non-Apple platforms such as BSD, Windows, Android and so on.

https://github.com/apple/swift/blob/c139bb4a1c81f8c39d0f0ccd4f7efc3c8131bed1/lib/Basic/LangOptions.cpp#L42-L57

---

In Quick `canImport(Darwin)` is used widely: https://github.com/Quick/Quick/search?q=canImport+Darwin